src/layers/domain/core/cpm_product/v1.py

```diff
@@ -62,3 +80,13 @@ def add_key(self, key_type: str, key_value: str) -> CpmProductKeyAddedEvent:
         product_data = self.state()
         product_data.pop(UPDATED_ON)  # The @event decorator will handle updated_on
         return CpmProductKeyAddedEvent(new_key=product_key, **product_data)
+
+    @event
+    def delete(self):
+        deleted_on = now()
+        product_data = self._update(
+            data=dict(
+                status=Status.INACTIVE, updated_on=deleted_on, deleted_on=deleted_on
+            )
+        )
+        return CpmProductDeletedEvent(**product_data)
```

Comment:
Where do we set the pk and sk to like P#INACTIVE? (I tried to find where we were doing it previously for devices but also got lost so this is just for my understanding aha)

Reply:
ah yes very good shout